docs/sources/flow/reference/components/discovery.kubelet.md

```diff
@@ -0,0 +1,157 @@
+`node` | `bool`   | Attach node metadata. | | no
+
+### authorization block
+
+{{< docs/shared lookup="flow/reference/components/authorization-block.md" source="agent" >}}
+
+### tls_config block
+
+{{< docs/shared lookup="flow/reference/components/tls-config-block.md" source="agent" >}}
+
+## Exported fields
+
+The following fields are exported and can be referenced by other components:
+
+Name | Type | Description
+---- | ---- | -----------
+`targets` | `list(map(string))` | The set of targets discovered from the Kubelet API.
+
+Each target includes the following labels::
+
+* `__meta_kubernetes_namespace`: The namespace of the pod object.
+* `__meta_kubernetes_pod_name`: The name of the pod object.
+* `__meta_kubernetes_pod_ip`: The pod IP of the pod object.
+* `__meta_kubernetes_pod_label_<labelname>`: Each label from the pod object.
+* `__meta_kubernetes_pod_labelpresent_<labelname>`: `true` for each label from
+  the pod object.
+* `__meta_kubernetes_pod_annotation_<annotationname>`: Each annotation from the
+  pod object.
+* `__meta_kubernetes_pod_annotationpresent_<annotationname>`: `true` for each
+  annotation from the pod object.
+* `__meta_kubernetes_pod_container_init`: `true` if the container is an
+  `InitContainer`.
+* `__meta_kubernetes_pod_container_name`: Name of the container the target
+  address points to.
+* `__meta_kubernetes_pod_container_id`: ID of the container the target address
+  points to. The ID is in the form `<type>://<container_id>`.
+* `__meta_kubernetes_pod_container_image`: The image the container is using.
+* `__meta_kubernetes_pod_container_port_name`: Name of the container port.
+* `__meta_kubernetes_pod_container_port_number`: Number of the container port.
+* `__meta_kubernetes_pod_container_port_protocol`: Protocol of the container
+  port.
+* `__meta_kubernetes_pod_ready`: Set to `true` or `false` for the pod's ready
+  state.
+* `__meta_kubernetes_pod_phase`: Set to `Pending`, `Running`, `Succeeded`, `Failed` or
+  `Unknown` in the lifecycle.
+* `__meta_kubernetes_pod_node_name`: The name of the node the pod is scheduled
+  onto.
+* `__meta_kubernetes_pod_host_ip`: The current host IP of the pod object.
+* `__meta_kubernetes_pod_uid`: The UID of the pod object.
+* `__meta_kubernetes_pod_controller_kind`: Object kind of the pod controller.
+* `__meta_kubernetes_pod_controller_name`: Name of the pod controller.
+
+## Component health
+
+`discovery.kubelet` is reported as unhealthy when given an invalid
+configuration. In those cases, exported fields retain their last healthy
+values.
+
+## Debug information
+
+`discovery.kubelet` does not expose any component-specific debug information.
+
+### Debug metrics
+
+`discovery.kubelet` does not expose any component-specific debug metrics.
+
+## Examples
+
+### Bearer token file authentication
+
+This example uses a bearer token file to authenticate to the Kubernetes API:
+
+```river
+discovery.kubernetes "k8s_pods" {
```

Comment:
Same here with needs to be `discovery.kubelet`

Reply:
Oops too much copy and paste, fixed!